### ISSUE TITLE
Fix target_link_libraries

### ISSUE THREAD

**Issue by JoseTomasTocino** [erikzenker/inotify-cpp#55]
When the library is used as an imported target, like the following:

```
add_subdirectory(inotify-cpp)
add_executable(MonitorPlainCpp main.cpp)
target_link_libraries(MonitorPlainCpp inotify-cpp)
```

There are linker errors like the following:

```
/usr/bin/ld: CMakeFiles/MonitorPlainCpp.dir/main.cpp.o: undefined reference to symbol '_ZN5boost6system15system_categoryEv'
/usr/lib64/libboost_system-mt.so.1.53.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```

That is due to the `inotify` libraries not being linked properly. [This line](https://github.com/erikzenker/inotify-cpp/blob/master/src/CMakeLists.txt#L49), that says:

```
target_link_libraries(${LIB_NAME}
        PRIVATE
          Boost::system
          Boost::filesystem)
```

should be

```
target_link_libraries(${LIB_NAME}
        INTERFACE
          Boost::system
          Boost::filesystem)
```

**Comment by erikzenker** [comment]
Hi Jose, thx for the hint. I will work on a fix!

**Comment by erikzenker** [comment]
Fixed with #56
